### PR TITLE
Travis: test on os x

### PR DESCRIPTION
**Pull request by LinusU**
This instructs Travis CI to run the tests on both OS X and on Linux.

### CODE REVIEW

**Comment by janisozaur** [comment]
the problem with osx on travis is that it rebuilds packages on every run, this is hugely inefficient and takes really long time. I managed to get "bottled" versions of required packages prepared, we should be using these instead.

I'll try to dig them up, but you probably know brew better then me, so I'd like to ask you to include them in the build.

**Comment by LinusU** [comment]
Hmm, homebrew itself only supplies bottles for formulas installed without options. Since we need to install them with `--universal` (to get 32-bit build as well) we can't use the official bottles. I don't know how it would work providing our own bottles, will have to look into that...

At least I'm getting an interesting error now:

> /Users/travis/build/OpenRCT2/OpenRCT2/src/hook.c:232:91: error: use of undeclared identifier 'MAP_ANONYMOUS'

It seems like it's called `MAP_ANON` on BSD (and OS X), I wonder why this compiles locally for me...

**Comment by janisozaur** [comment]
aren't you using my `osx` branch locally?

**Comment by LinusU** [comment]
Nope, I'm building from the latest `develop` now. Although I still have the problem with cmake and SDL2_ttf that we discussed...

I'm using this to work around it:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3eba820..4e79525 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,7 +68,11 @@ if (UNIX)
 endif (UNIX)
 
 # find and include SDL2
-PKG_CHECK_MODULES(SDL2 REQUIRED sdl2 SDL2_ttf)
+PKG_CHECK_MODULES(SDL2 REQUIRED sdl2)
+PKG_CHECK_MODULES(SDL2_TTF REQUIRED SDL2_ttf)
+
+# find and include iconv
+FIND_LIBRARY(ICONV_LIBRARIES NAMES libiconv)
 
 # speex v1.1.15 is supplied in our zipped library, but distributions provide
 # updated version, with required functions extracted out to libspeexdsp.
@@ -85,9 +89,9 @@ if (UNIX)
 	set(DLLIB dl)
 endif (UNIX)
 
-INCLUDE_DIRECTORIES(${SDL2_INCLUDE_DIRS} ${LIBCURL_INCLUDE_DIRS} ${JANSSON_INCLUDE_DIRS} ${SPEEX_INCLUDE_DIRS})
+INCLUDE_DIRECTORIES(${SDL2_INCLUDE_DIRS} ${SDL2_TTF_INCLUDE_DIRS} ${LIBCURL_INCLUDE_DIRS} ${JANSSON_INCLUDE_DIRS} ${SPEEX_INCLUDE_DIRS})
 
-LINK_DIRECTORIES(${SDL2_LIBRARY_DIRS} ${JANSSON_LIBRARY_DIRS} ${LIBCURL_LIBRARY_DIRS})
+LINK_DIRECTORIES(${SDL2_LIBRARY_DIRS} ${SDL2_TTF_LIBRARY_DIRS} ${JANSSON_LIBRARY_DIRS} ${LIBCURL_LIBRARY_DIRS})
 
 if (WIN32)
 	# build as library for now, replace with add_executable
@@ -102,4 +106,4 @@ endif (WIN32)
 # libopenrct2.dll -> openrct2.dll
 set_target_properties(${PROJECT} PROPERTIES PREFIX "")
 
-TARGET_LINK_LIBRARIES(${PROJECT} ${SDL2_LIBRARIES} ${ORCTLIBS_LIB} ${JANSSON_LIBRARIES} ${HTTPLIBS} ${NETWORKLIBS} ${SPEEX_LIBRARIES} ${DLLIB})
+TARGET_LINK_LIBRARIES(${PROJECT} ${SDL2_LIBRARIES} ${SDL2_TTF_LIBRARIES} ${ICONV_LIBRARIES} ${ORCTLIBS_LIB} ${JANSSON_LIBRARIES} ${HTTPLIBS} ${NETWORKLIBS} ${SPEEX_LIBRARIES} ${DLLIB})
```

**Comment by LinusU** [comment]
So at least `SDL2_ttf` works on Travis, that's nice. It seems like the iconv fix is needed though, adding that now.

**Comment by janisozaur** [comment]
`iconv` is part of `glibc`, so you'll have to wrap 3596684815ed in some kind of check for OS X, or better yet, do a compile-time check.
https://travis-ci.org/janisozaur/OpenRCT2/builds/93890558

**Comment by LinusU** [comment]
Hmm, gave it another try: a7bceb7

**Comment by janisozaur** [comment]
https://travis-ci.org/janisozaur/OpenRCT2/builds/93896990

**Comment by LinusU** [comment]
So close :)

I'm at work now so I'll have to look at it tonight...

**Comment by LinusU** [comment]
@Gymnasiast ready for review :fireworks:

**Comment by LinusU** [comment]
I've only added one OS X test since they take a bit longer to run than the linux ones. I figured this was a good tradeoff between starting testing on OS X as well and also not increasing the test time too much. I don't think overall build time increased that much since your allowed a few number of parallel builds.

As noted in other issues, OS X support isn't quite ready for prime time yet. But I believe that we are getting very close!

File: install.sh [hunk 1]
```diff
@@ -102,66 +106,50 @@ function install_local_libs {
...
+	echo "Installation of OpenRCT2 assumes you have homebrew and use it to install packages."
+
+	if ! has_cmd "brew"; then
+		echo "brew is not installed, or is not in your \$PATH"
+		echo "install instructions: http://brew.sh/"
+		exit 1
+	fi
+
+	# Very possible I'm missing some dependencies here.
+	brew install cmake
```
Comment (janisozaur) [inline install.sh]: Can these be merged to be just two lines (for `--universal` and regular)?

File: install.sh [hunk 2]
```diff
@@ -102,66 +106,50 @@ function install_local_libs {
 	cp -rf $cachedir/orctlibs/local/* ./lib/.
 }
 
+function install_mingw_32 {
```
Comment (janisozaur) [inline install.sh]: This is only for OSX for now, so this function should either handle all supported cases or has a name that reflects its true nature.

File: .travis.yml [hunk 3]
```diff
@@ -1,34 +1,49 @@
...
+    - .cache
+  apt: true
 
 sudo: required
 dist: trusty
 
-services:
-    - docker
+matrix:
+  include:
```
Comment (janisozaur) [inline .travis.yml]: This is becoming quite a huge testing matrix, which I don't think we really need. It should be refactored and possibly trimmed down.

File: install.sh [hunk 4]
```diff
@@ -15,10 +15,14 @@ TARGET=${TARGET-linux}
 libversion=3
 libVFile="./libversion"
 
+function has_cmd {
+	command -v "$1" >/dev/null 2>&1
```
Comment (janisozaur) [inline install.sh]: is there `which` on OS X? I'd rather you used that instead of `command -v`

**Comment by janisozaur** [comment]
I had a look at that and I don't think it's ready yet.
I made a few comments, but generally speaking:
* I don't like the packages being built every time we have a commit. This is terribly wasteful and gets a huge :-1: from me. Please investigate bottling `--universal` packages properly, caching them and pouring from bottles when doing tests (that's some crazy brew-speak :cocktail:). It should be easy enough for you to do, I've almost done that without access to OS X. It takes 288s to build & install deps, while only 25s to test openrct.
* Can you include testing g2 generation as well?
* Build matrix starts getting out of hand, this has to be improved.

File: install.sh (hunk 4 above)
Comment (LinusU) [inline install.sh]: Why? `command -v` was also used before in the script and I thought that it was best practice.

see e.g. http://stackoverflow.com/questions/592620/check-if-a-program-exists-from-a-bash-script

File: install.sh (hunk 2 above)
Comment (LinusU) [inline install.sh]: True, I'll update the name to `os_x_install_mingw_32`, sounds good?

File: install.sh (hunk 1 above)
Comment (LinusU) [inline install.sh]: Seems like it, I'll update

File: install.sh (hunk 4 above)
Comment (janisozaur) [inline install.sh]: I don't like that we blindly assume that `command -v` is harmless.

But with your link it's OK to leave it as is for our limited usage.

File: install.sh (hunk 2 above)
Comment (janisozaur) [inline install.sh]: ok

**Comment by janisozaur** [comment]
What version of OS X are you running? Can you check https://github.com/janisozaur/OpenRCT2/tree/osx-linusu ?

**Comment by LinusU** [comment]
Latest greatest, 10.11.1 El Capitan!

It actually opens up a black window, but segfaults shortly after.

```text
WARNING[OpenRCT2/src/platform/posix.c:54 (main)]: Function int main(int, const char **) at OpenRCT2/src/platform/posix.c:54 is a stub.

WARNING[OpenRCT2/src/openrct2.c:534 (openrct2_setup_rct2_segment)]: mmapped file to 0x1429000
WARNING[OpenRCT2/src/util/util.c:209 (safe_strncpy)]: Truncating string  to 35 bytes.
WARNING[OpenRCT2/src/platform/posix.c:729 (platform_get_locale_currency)]: Function uint8 platform_get_locale_currency() at OpenRCT2/src/platform/posix.c:729 is a stub.

WARNING[OpenRCT2/src/platform/posix.c:676 (platform_get_locale_language)]: Function uint16 platform_get_locale_language() at OpenRCT2/src/platform/posix.c:676 is a stub.

WARNING[OpenRCT2/src/platform/posix.c:749 (platform_get_locale_measurement_format)]: Function uint8 platform_get_locale_measurement_format() at OpenRCT2/src/platform/posix.c:749 is a stub.

WARNING[OpenRCT2/src/platform/posix.c:771 (platform_get_locale_temperature_format)]: Function uint8 platform_get_locale_temperature_format() at OpenRCT2/src/platform/posix.c:771 is a stub.

WARNING[OpenRCT2/src/rct2.c:530 (get_system_info)]: Function void get_system_info() at OpenRCT2/src/rct2.c:530 is a stub.

WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: /Users/linus/.config/OpenRCT2/themes/*.ini
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.INI
WARNING[OpenRCT2/src/platform/posix.c:251 (platform_enumerate_files_begin)]: found 0 files matching in dir '/Users/linus/.config/OpenRCT2/themes'
WARNING[OpenRCT2/src/platform/posix.c:104 (platform_file_exists)]: file 'OpenRCT2/build/data/title/rct2/script.txt' exists = 1
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/data/title/rct2/*.sv6
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.SV6
WARNING[OpenRCT2/src/platform/posix.c:251 (platform_enumerate_files_begin)]: found 0 files matching in dir 'OpenRCT2/build/data/title/rct2'
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/data/title/rct2/*.sc6
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.SC6
WARNING[OpenRCT2/src/platform/posix.c:251 (platform_enumerate_files_begin)]: found 0 files matching in dir 'OpenRCT2/build/data/title/rct2'
WARNING[OpenRCT2/src/platform/posix.c:104 (platform_file_exists)]: file 'OpenRCT2/build/data/title/openrct2/script.txt' exists = 1
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/data/title/openrct2/*.sv6
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.SV6
WARNING[OpenRCT2/src/platform/posix.c:251 (platform_enumerate_files_begin)]: found 11 files matching in dir 'OpenRCT2/build/data/title/openrct2'
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/data/title/openrct2/*.sc6
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.SC6
WARNING[OpenRCT2/src/platform/posix.c:251 (platform_enumerate_files_begin)]: found 0 files matching in dir 'OpenRCT2/build/data/title/openrct2'
WARNING[OpenRCT2/src/platform/posix.c:377 (platform_enumerate_directories_begin)]: begin directory listing, path: /Users/linus/.config/OpenRCT2/title sequences/
WARNING[OpenRCT2/src/util/util.c:209 (safe_strncpy)]: Truncating string openrct2/*.sc6 to 46 bytes.
WARNING[OpenRCT2/src/platform/posix.c:391 (platform_enumerate_directories_begin)]: found 0 files in dir '/Users/linus/.config/OpenRCT2/title sequences/'
WARNING[OpenRCT2/src/platform/posix.c:104 (platform_file_exists)]: file 'OpenRCT2/build/Data/g1.dat' exists = 1
WARNING[OpenRCT2/src/rct2.c:372 (check_files_integrity)]: Function int check_files_integrity() at OpenRCT2/src/rct2.c:372 is a stub.

WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/ObjData/*.DAT
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.DAT
WARNING[OpenRCT2/src/platform/posix.c:251 (platform_enumerate_files_begin)]: found 772 files matching in dir 'OpenRCT2/build/ObjData'
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/ObjData/*.DAT
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.DAT
WARNING[OpenRCT2/src/platform/posix.c:251 (platform_enumerate_files_begin)]: found 772 files matching in dir 'OpenRCT2/build/ObjData'
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/Tracks/*.TD?
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.TD?
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/Tracks/*.TD?
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.TD?
ERROR[OpenRCT2/src/scenario_list.c:188 (scenario_scores_load)]: Unable to load scenario scores.
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/Scenarios/*.SC6
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.SC6
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/Tracks/*.TD?
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.TD?
ERROR[OpenRCT2/src/ride/track.c:314 (track_list_cache_load)]: Failed to load /Users/linus/.config/OpenRCT2/tracks.idx
WARNING[OpenRCT2/src/platform/posix.c:220 (platform_enumerate_files_begin)]: begin file search, pattern: OpenRCT2/build/Tracks/*.TD?
WARNING[OpenRCT2/src/platform/posix.c:240 (platform_enumerate_files_begin)]: looking for file matching *.TD?
FATAL[OpenRCT2/src/drawing/sprite.c:110 (gfx_load_g2)]: Unable to load g2 graphics
WARNING[OpenRCT2/src/platform/shared.c:877 (platform_load_cursors)]: Function void platform_load_cursors() at OpenRCT2/src/platform/shared.c:877 is a stub.

WARNING[OpenRCT2/src/platform/osx.c:28 (platform_check_steam_overlay_attached)]: Function _Bool platform_check_steam_overlay_attached() at OpenRCT2/src/platform/osx.c:28 is a stub.

WARNING[OpenRCT2/src/interface/window.c:2431 (textinput_cancel)]: there should be something called here (0x0040701D)
WARNING[OpenRCT2/src/interface/window.c:2431 (textinput_cancel)]: there should be something called here (0x0040701D)
[1]    64406 segmentation fault  build/openrct2
```

**Comment by LinusU** [comment]
As I saw discussed in another thread, the program crashes even more early if run thru lldb.

```text
(lldb) target create "build/openrct2"
Current executable set to 'build/openrct2' (i386).
(lldb) r
Process 64447 launched: 'OpenRCT2/build/openrct2' (i386)
WARNING[OpenRCT2/src/platform/posix.c:54 (main)]: Function int main(int, const char **) at OpenRCT2/src/platform/posix.c:54 is a stub.

WARNING[OpenRCT2/src/openrct2.c:534 (openrct2_setup_rct2_segment)]: mmapped file to 0x2100000
WARNING[OpenRCT2/src/util/util.c:209 (safe_strncpy)]: Truncating string  to 35 bytes.
WARNING[OpenRCT2/src/platform/posix.c:729 (platform_get_locale_currency)]: Function uint8 platform_get_locale_currency() at OpenRCT2/src/platform/posix.c:729 is a stub.

WARNING[OpenRCT2/src/platform/posix.c:676 (platform_get_locale_language)]: Function uint16 platform_get_locale_language() at OpenRCT2/src/platform/posix.c:676 is a stub.

WARNING[OpenRCT2/src/platform/posix.c:749 (platform_get_locale_measurement_format)]: Function uint8 platform_get_locale_measurement_format() at OpenRCT2/src/platform/posix.c:749 is a stub.

WARNING[OpenRCT2/src/platform/posix.c:771 (platform_get_locale_temperature_format)]: Function uint8 platform_get_locale_temperature_format() at OpenRCT2/src/platform/posix.c:771 is a stub.

Process 64447 stopped
* thread #1: tid = 0x22204e, 0x9057b71f libobjc.A.dylib`NXHashGet + 30, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x41e9d815)
    frame #0: 0x9057b71f libobjc.A.dylib`NXHashGet + 30
libobjc.A.dylib`NXHashGet:
->  0x9057b71f <+30>: calll  *(%eax)
    0x9057b721 <+32>: movl   %edi, %ecx
    0x9057b723 <+34>: xorl   %edx, %edx
    0x9057b725 <+36>: divl   0x8(%ecx)
(lldb) bt
* thread #1: tid = 0x22204e, 0x9057b71f libobjc.A.dylib`NXHashGet + 30, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x41e9d815)
  * frame #0: 0x9057b71f libobjc.A.dylib`NXHashGet + 30
    frame #1: 0x9057e45a libobjc.A.dylib`look_up_class + 82
    frame #2: 0x9057ed46 libobjc.A.dylib`objc_getClass + 37
    frame #3: 0x9057ce1d libobjc.A.dylib`_class_getNonMetaClass + 99
    frame #4: 0x9057cc4f libobjc.A.dylib`lookUpImpOrForward + 110
    frame #5: 0x9057cbdc libobjc.A.dylib`_class_lookupMethodAndLoadCache3 + 55
    frame #6: 0x9057c704 libobjc.A.dylib`objc_msgSend + 132
    frame #7: 0x000a5e17 libSDL2-2.0.0.dylib`SDL_OpenFPFromBundleOrFallback + 14
    frame #8: 0x0003a6fd libSDL2-2.0.0.dylib`SDL_RWFromFile_REAL + 52
    frame #9: 0x03408d19 openrct2`config_open + 41
    frame #10: 0x03408c84 openrct2`config_open_default + 132
    frame #11: 0x03455956 openrct2`openrct2_initialise + 294
    frame #12: 0x03455ce4 openrct2`openrct2_launch + 20
    frame #13: 0x03474ac0 openrct2`main + 176
    frame #14: 0x9aed26ad libdyld.dylib`start + 1
(lldb) 
```

**Comment by janisozaur** [comment]
there should be now one more commit, which shouldn't affect anything, but it is what was different in my version from what i pushed. perhaps i missed something here by chance, so just to be sure.

also, i was using xcode 7.1 and specified build type `debug`.

have you wired up all the dirs correctly? `data`, `Data`, `ObjData` are the least required.

does that get any further?

**Comment by LinusU** [comment]
The data dirs should be wired up correctly, just got to work so I'll check tonight :ok_hand:

**Comment by janisozaur** [comment]
grab `g2.dat` from Linux release on github and it should work :)

**Comment by AaronVanGeffen** [comment]
Confirmed. OpenRCT2 now running natively on OS X 10.11 here. Graphical glitches due to hooks not working properly, but very exciting progress!

**Comment by LinusU** [comment]
I copied in `g2.dat` but now I get the following stack trace:

```text
Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x00000000132ebd38

VM Regions Near 0x132ebd38:
    VM_ALLOCATE            000000000a957000-000000000a958000 [    4K] rw-/rwx SM=ALI  
--> 
    MALLOC_TINY            0000000078600000-0000000078800000 [ 2048K] rw-/rwx SM=PRV  

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   openrct2                      	0x0341e109 update_palette_effects + 905
1   openrct2                      	0x03482d9d rct2_draw + 45
2   openrct2                      	0x0345ab7b openrct2_loop + 1403
3   openrct2                      	0x03459e46 openrct2_launch + 374
4   openrct2                      	0x03478ac0 main + 176
5   libdyld.dylib                 	0x9aed26ad start + 1
```

I will try it with the other commented out malloc now...

**Comment by LinusU** [comment]
No luck :( on the latest `osx-linusu` branch now.

```text
WARNING[/Users/linus/coding/OpenRCT2/src/util/util.c:209 (safe_strncpy)]: Truncating string  to 51 bytes.
2015-12-07 21:57:24.595 openrct2[65113:2405071] 21:57:24.594 WARNING:  140: This application, or a library it uses, is using the deprecated Carbon Component Manager for hosting Audio Units. Support for this will be removed in a future release. Also, this makes the host incompatible with version 3 audio units. Please transition to the API's in AudioComponent.h.
[1]    65113 segmentation fault  build/openrct2
```

**Comment by janisozaur** [comment]
will `libpng` and `xz` get installed in universal version with latest commit?

**Comment by LRFLEW** [comment]
I'm having issues with getting this to work. I'm getting an `EXC_BAD_ACCESS` error at `update_palette_effects + 905 (game.c:181)`. Using `printf`'s, I was able to determine that the crash is occurring because the variable `p` is getting set to the ridiculous value of `-1322713087` at line `game.c:179`.

**Comment by LinusU** [comment]
@LRFLEW Thanks for that info! After commenting out that line (`p = water_type->var_06;`) the game starts!

**Comment by LinusU** [comment]
I will rebase this once #2469 has landed and then try to finally fix the damn bottles :smile:

**Comment by janisozaur** [comment]
@LinusU any progress with these bottles?

**Comment by LinusU** [comment]
Hopefully I'll get time to look at it this weekend, sorry for the delay

**Comment by janisozaur** [comment]
With #2518 now merged, you should switch to XCode project instead of using CMake.

**Comment by LinusU** [comment]
Yes! No more homebrewed bottles :D

**Comment by janisozaur** [comment]
Huh? I thought you still need them?

**Comment by LinusU** [comment]
Thru `orctlibs-mac.zip`, I think it includes prebuilt libs :)

**Comment by janisozaur** [comment]
Oh, I missed that. OK, can you update this PR then?

**Comment by LinusU** [comment]
Yes, I'll do that :)

**Comment by janisozaur** [comment]
Superseded by https://github.com/OpenRCT2/OpenRCT2/pull/2669

**Pull request closed without merging**